cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/instance/AwsNativeInstanceResourceBuilder.java

```diff
@@ -176,8 +178,14 @@ private boolean instanceTerminated(Instance instance) {
         return instance.getState().getCode() == AWS_INSTANCE_TERMINATED_CODE;
     }
 
-    private String getUserData() {
-        return "";
+    private String getUserData(CloudStack cloudStack, Group group) {
+        String userdata;
+        if (group.getType() == InstanceGroupType.CORE) {
+            userdata = cloudStack.getImage().getUserDataByType(InstanceGroupType.CORE);
+        } else {
+            userdata = cloudStack.getImage().getUserDataByType(InstanceGroupType.GATEWAY);
+        }
```

Comment:
Can be simplified to `userdata = cloudStack.getImage().getUserDataByType(group.getType());`, at least if no further enum constants are ever expected in `InstanceGroupType`.